contracts/modules/v2/StakingPool.sol

```diff
@@ -747,15 +759,53 @@ contract StakingPool is IStakingPool, ERC721 {
   /* nft */
 
   function _beforeTokenTransfer(
-    address from,
+    address /*from*/,
     address /*to*/,
     uint256 tokenId
   ) internal view override {
-    require(
-      tokenId != 0 || nxm.isLockedForMV(from) < block.timestamp,
-      "StakingPool: Locked for voting in governance"
-    );
-    // todo: track owned zero-id nfts in TC
+    if(tokenId == 0) {
+      require(
+        nxm.isLockedForMV(manager()) < block.timestamp,
+        "StakingPool: Active pool assets are locked for voting in governance"
+      );
+    }
+  }
+
+  /**
+   * @dev See {IERC721-transferFrom}.
+   */
+  function transferFrom(
+      address from,
+      address to,
+      uint256 tokenId
+  ) public override(ERC721, IERC721) {
+      _beforeTokenTransfer(from, to, tokenId);
+      super.transferFrom(from, to, tokenId);
+  }
+
+  /**
+   * @dev See {IERC721-safeTransferFrom}.
+   */
+  function safeTransferFrom(
+      address from,
+      address to,
+      uint256 tokenId
+  ) public override(ERC721, IERC721) {
+      _beforeTokenTransfer(from, to, tokenId);
+      super.safeTransferFrom(from, to, tokenId);
+  }
+
+  /**
+   * @dev See {IERC721-safeTransferFrom}.
+   */
+  function safeTransferFrom(
+      address from,
+      address to,
+      uint256 tokenId,
+      bytes memory _data
+  ) public override(ERC721, IERC721) {
+      _beforeTokenTransfer(from, to, tokenId);
+      super.safeTransferFrom(from, to, tokenId, _data);
```

Comment:
You correctly identified the fact that the hooked is not invoked, my bad. However the fix is to call `super._beforeTokenTransfer(from, to, tokenId);` inside the `_beforeTokenTransfer` function, not calling it in all transfer functions. https://docs.openzeppelin.com/contracts/4.x/extending-contracts#rules_of_hooks